src/main/java/sg/ncl/webssh/SentOutputTask.java

```diff
@@ -0,0 +1,54 @@
+package sg.ncl.webssh;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Scope;
+import org.springframework.messaging.simp.SimpMessagingTemplate;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.Charset;
+
+/**
+ * References:
+ * [1] https://www.mkyong.com/spring/spring-and-java-thread-example/
+ * [2] http://codenav.org/code.html?project=/com/nitorcreations/willow-servers/0.1&path=/Source%20Packages/com.nitorcreations.willow.ssh/SecureShellWS.java
+ * [3] https://stackoverflow.com/questions/37307697/scheduled-websocket-push-with-springboot
+ */
+
+@Component
+@Scope("prototype")
+@Slf4j
+public class SentOutputTask implements Runnable {
+
+    static final int BUFFER_LEN = 4 * 1024;
+
+    @Autowired
+    private SimpMessagingTemplate template;
+
+    private InputStream output;
+
+    public void setOutput(InputStream output) {
+        this.output = output;
+    }
+
+    @Override
+    public void run() {
+        byte[] buf = new byte[BUFFER_LEN];
+        int read;
+
+        try {
+            while ((read = output.read(buf)) != -1) {
+                if (read > 0) {
+                    template.convertAndSend("/shell/output", new String(buf, 0, read, Charset.forName("UTF-8")));
+                }
+                Thread.sleep(100);
+            }
+        } catch (IOException e) {
+            log.info("Failed to read data from ssh: {}", e);
+        } catch (InterruptedException e) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either re-interrupt this method or rethrow the "InterruptedException". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2142)